### DISCUSSION TITLE
Reduce the number of requests to the registry

### DISCUSSION THREAD

**Discussion by ngosang** [crazy-max/diun#860] [Q&A] [answered]
I have been testing this software and it's promising. Crongrats!
My only complain is it makes a lot of requests to the registry.
I have this docker image => https://hub.docker.com/r/linuxserver/jackett/tags
As you can see they publish 20+ new tags every day.
If I set "watch_repo=false", Diun works fine because it only gets the metadata for the "latest" tag.
My goal is to set "watch_repo=true" and use "include_tags" to filter only major releases. This configuration is overkill because even if i set "max_tags=20" it pulls metadata for a lot of images.

Is it possible to request only the list of tags and skip the metadata part? I just need the "new tag name" notification.

**Comment by crazy-max** [comment, marked as answer]
With `watch_repo=true`, Diun will make one request to fetch all tags and one request per tag to fetch the manifest. If you use `watch_repo=true` and `include_tags` it will do one request to fetch all tags but will fetch manifest for only the tags included by `include_tags`.

Maybe your regexp for `include_tags` is wrong and don't filter anything. You can take a look at some examples in https://crazymax.dev/diun/providers/file/#example

If you still think there is an issue, please create a proper [bug report](https://github.com/crazy-max/diun/issues/new?assignees=&labels=&template=bug_report.md). Thanks.

**Comment by fredgate** [comment]
I'm looking to use diun configured with a list of repositories on Docker hub in the file provider, and to receive notifications when a new image is available.
So I don't understand why diun needs to pull the relevant images locally? Isn't retrieving the list of tags enough?
